### PR TITLE
Add TextInputType.none

### LINKED ISSUE #83567: Add TextInputType.none
## Use case

It would be great to be able to disable the virtual keyboard for certain TextFields.

## Proposal

Introduce `TextInputType.none` that can be specified as the keyboard type for TextField. Flutter should not open the virtual keyboard when such text field gains focus.

In order to avoid the platform keyboard popping up, Flutter for Web needs to set the input element's [`inputmode`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode) attribute to `none`. This matches well with the [`TextInputType`](https://api.flutter.dev/flutter/services/TextInputType-class.html) enum that is already used to define the attribute.

Similarly, the rest of the platforms need to respect `TextInputType.none` when passed in the text input configuration.

## Additional notes

This would lay the foundations for custom in-app virtual keyboards: #68988 and #76072.

PR: #83974

### PR DESCRIPTION
**Pull request by jpnurmi**
`TextInputType.none` makes it possible to disable the virtual keyboard for individual TextFields, and lays the foundations for custom in-app virtual keyboards (flutter/flutter#76072).

This is just the framework part that merely introduces the `TextInputType.none` enum value. For the actual functionality, the engine needs to be adapted for all platforms to respect `TextInputType.none` when passed in the text input configuration.
- [x] Android (https://github.com/flutter/engine/pull/26585)
- [x] iOS (https://github.com/flutter/engine/pull/26979)
- [x] Linux (https://github.com/flutter/engine/pull/26574)
- [ ] ~~macOS~~
- [x] Web (https://github.com/flutter/engine/pull/26573)
- [ ] ~~Windows~~

Closes: flutter/flutter#83567

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/services/text_input.dart [hunk 1]
```diff
@@ -174,14 +174,17 @@ class TextInputType {
   /// [TYPE_TEXT_VARIATION_POSTAL_ADDRESS](https://developer.android.com/reference/android/text/InputType#TYPE_TEXT_VARIATION_POSTAL_ADDRESS).
   static const TextInputType streetAddress = TextInputType._(9);
 
+  /// No keyboard.
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/services/text_input.dart:177]: nit: 
```suggest
/// Prevent the OS from showing the on-screen virtual keyboard.
```

**Review by LongCatIsLooong — APPROVED** [review]
LGTM. I'll hold off until the engine changes are LGTM'd.

**Comment by jpnurmi** [comment]
I suppose there's not much we can do on Windows at this point. `TextInput.show` and `TextInput.hide` do nothing: https://github.com/flutter/engine/blob/2384128a226c6661844c247a31204235372a81e6/shell/platform/windows/text_input_plugin.cc#L148. There seems to be a ticket about it: https://github.com/flutter/flutter/issues/36057

**Comment by jpnurmi** [comment]
On macOS, the [Accessibility Keyboard](https://support.apple.com/guide/mac-help/use-the-accessibility-keyboard-mchlc74c1c9f/mac) is a persistent floating top-level window, and its visibility is controlled by the user. So nothing to do on macOS either?

**Review by justinmc — APPROVED** [review]
LGTM 👍 

Looks like the analyzer is down or something?  It's not able to run.

Edit: Windows and Mac seem ok to me also.  I would think it's desired that the user can still control an accessibility keyboard even if a Mac app has implemented a custom virtual keyboard using this feature.

**Comment by LongCatIsLooong** [comment]
@jpnurmi Could you try merging/rebasing the branch to the tip of the tree?

**Comment by ericHandtke** [comment]
Is it possible to use this yet?

**Comment by ivanmem** [comment]
With autofocus false the text is also not entered.
image

**Pull request merged**